### PR TITLE
feat: finish migrating schema extraction!!!!

### LINKED ISSUE #1518: refactor go-runtime schema extraction
this file has gotten out of hand!!!!
it's quite difficult to reason through

refactor will split out schema concepts into separate scans (e.g. parse type enums, parse value enums, parse verbs, etc.)

reference https://pkg.go.dev/golang.org/x/tools/go/analysis

### PR DESCRIPTION
**Pull request by worstell**
- migrates FSMs to new extractor
- deletes legacy code

fixes #1518

### CODE REVIEW

**Comment by alecthomas** [comment]
Yayyyy! Epic job Lizzy, nice one 😀

File: ftl-project.toml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 name = "ftl"
 module-dirs = ["examples/go"]
 ftl-min-version = ""
+hermit = false
+no-git = false
```
Comment (wesbillman) [inline ftl-project.toml:5]: are these required for the schema extraction changes? Similar to the `modules.test` below.

File: go-runtime/schema/common/common.go [hunk 2]
```diff
@@ -67,6 +67,15 @@ func NewDeclExtractor[T schema.Decl, N ast.Node](name string, extractFunc Extrac
...
 }
 
+// ExtractCallDeclFunc extracts a schema declaration from the given node.
+type ExtractCallDeclFunc[T schema.Decl] func(pass *analysis.Pass, object types.Object, node *ast.GenDecl, callExpr *ast.CallExpr, callPath string) optional.Option[T]
+
+// NewCallDeclExtractor extracts declarations from call expressions, e.g. `ftl.Subscription("name")`.
+func NewCallDeclExtractor[T schema.Decl](name string, extractFunc ExtractCallDeclFunc[T], callPaths ...string) *analysis.Analyzer {
+	type Tag struct{} // Tag uniquely identifies the fact type for this extractor.
+	return NewExtractor(name, (*DefaultFact[Tag])(nil), runExtractCallDeclsFunc[T](extractFunc, callPaths...))
+}
```
Comment (wesbillman) [inline go-runtime/schema/common/common.go:77]: Nice!

**Review by wesbillman — APPROVED** [review]
Let's gooooooooo!

File: ftl-project.toml (hunk 1 above)
Comment (worstell) [reply ftl-project.toml:5]: ah not sure where these got picked up, thanks for flagging. let me check them out

**Pull request merged**